scene/3d/collision_object_3d.cpp

```diff
@@ -279,13 +324,19 @@ void CollisionObject3D::shape_owner_remove_shape(uint32_t p_owner, int p_shape)
 	ERR_FAIL_COND(!shapes.has(p_owner));
 	ERR_FAIL_INDEX(p_shape, shapes[p_owner].shapes.size());
 
-	int index_to_remove = shapes[p_owner].shapes[p_shape].index;
+	ShapeData::ShapeBase s = shapes[p_owner].shapes[p_shape];
```

Comment:
Here you could use `const ShapeData::ShapeBase &s`. Not very critical in this scenario, but again it would easily avoid an unnecessary copy.